RIOT-OS/RIOT
# cpu/sam0_common: don't default to ztimer_periph_rtt backend

### Contribution description

The sam0 rtt busy loops for 180us every time an alarm is set or
the counter is read, this propagates and leads to timing errors
on ztimer_msec that are higher than > +-1msec.

@dylad @benpicco is this the case for other sam? I would recommend doing the
same if it is the case.

### Testing procedure

Surfaces while testing #17357, this now passes

`USEMODULE="evtimer_on_ztimer ztimer_no_periph_rtt" BOARD=samr21-xpro make -C tests/evtimer_msg/ flash test -j`

```
main(): This is RIOT! (Version: 2022.01-devel-1231-g9223e1-pr_sam0_common_no_periph_rtt)
Testing generic evtimer
This should list 2 items
ev #1 offset=1000
ev #2 offset=500
This should list 4 items
ev #1 offset=659
ev #2 offset=341
ev #3 offset=500
ev #4 offset=2454
Are the reception times of all 4 msgs close to the supposed values?
At   3745 ms received msg 0: "#2 supposed to be 3745"
.At   4086 ms received msg 1: "#0 supposed to be 4086"
.At   4586 ms received msg 2: "#1 supposed to be 4586"
.At   7040 ms received msg 3: "#3 supposed to be 7040"
.
All tests successful

```
